src/Unleash/DefaultUnleash.cs

```diff
@@ -202,6 +202,34 @@ private IStrategy GetStrategyOrUnknown(string strategy)
                 : UnknownStrategy;
         }
 
+        private bool EnabledByAnyStrategy(FeatureToggle featureToggle, UnleashContext enhancedContext)
+        {
+            foreach (var activationStrategy in featureToggle.Strategies)
+            {
+                var segments = new List<Segment>(activationStrategy.Segments.Count);
+                foreach (var segmentId in activationStrategy.Segments)
+                {
+                    var segment = services.ToggleCollection.Instance.GetSegmentById(segmentId);
+                    if (segment == null)
+                    {
+                        return false;
+                    }
+                    segments.Add(segment);
```

Comment:
I think this is probably fine but the segments list might get largish. IIRC a list in C# has a memory size of N * sizeof T where T is the generic the list holds. This might be a lot of memory at load in a fairly tight loop. Might be worth checking what this does the process memory when there are 10K segments

Reply:
Yeah that's a very good point, and exactly the type of feedback i was soliciting. I'm doing an IEnumerable rewrite of this, i'll see if i can run a few simple benchmarks of it

Reply:
Amazing, that sounds great! There are a few scripts in Unleash server in the /perf folder to load a bunch of data into the server for testing exactly this, they were pretty useful for testing this when building the feature on the other SDKs